graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/method/AbstractInputArgumentResolver.kt

```diff
@@ -44,6 +47,19 @@ abstract class AbstractInputArgumentResolver(inputObjectMapper: InputObjectMappe
         val argumentName = getArgumentName(parameter)
         val value = dfe.getArgument<Any?>(argumentName)
 
+        val kfunc = parameter.method?.kotlinFunction
+        if (kfunc != null) {
+            val parameterIdx = if (kfunc.parameters.first().kind == KParameter.Kind.INSTANCE) {
```

Comment:
What does this check mean?

Reply:
The first parameter to the `KFunction` is the instance parameter, e.g. the instance the method is being called on, so we need to adjust the parameter index based on that (essentially start the indexing after this instance parameter).